### PR TITLE
(packaging) Update Bolt package component gems

### PR DESCRIPTION
**Pull request by lucywyman**
This updates the version of gem dependencies of Bolt to their latest
versions. This PR is limited to gems that are only included in the Bolt
and pe-bolt-server packages.

### CODE REVIEW

**Comment by lucywyman** [comment]
Hm the failure looks to be failing to wget from artifactory.

File: configs/components/rubygem-bindata.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 component 'rubygem-bindata' do |pkg, settings, platform|
-  pkg.version '2.4.4'
-  pkg.md5sum '132d77d64bedd315acff3b28b1129877'
+  pkg.version '2.4.7'
+  pkg.md5sum '516d8371db693d375dcdc57e862aa930'
```
Comment (beechtom) [inline configs/components/rubygem-bindata.rb:3]: md5sum mismatch. Should be `5803f4525dc4e2687b97aa37f0602a08`.

File: configs/components/rubygem-facter.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 component 'rubygem-facter' do |pkg, settings, platform|
-  pkg.version '4.0.13'
-  pkg.md5sum 'b75e5f9a0f13bfeec9c7e7b29a7e6625'
+  pkg.version '4.0.14'
```
Comment (beechtom) [inline configs/components/rubygem-facter.rb:2]: Well that update to 4.0.13 didn't last long. :P

File: configs/components/rubygem-bindata.rb (hunk 1 above)
Comment (lucywyman) [reply configs/components/rubygem-bindata.rb]: Dang good catch!

**Comment by beechtom** [comment]
Built locally and installed on a fedora 30 vm:

```
[root@mammoth-spa /]# /opt/puppetlabs/bolt/bin/gem list

*** LOCAL GEMS ***

addressable (2.7.0)
aws-eventstream (1.0.3)
aws-partitions (1.292.0)
aws-sdk-core (3.92.0)
aws-sdk-ec2 (1.151.0)
aws-sigv4 (1.1.1)
bcrypt_pbkdf (1.0.0)
bigdecimal (default: 1.3.4)
bindata (2.4.7)
bolt (2.3.1)
builder (3.2.4)
CFPropertyList (2.3.6)
cmath (default: 1.0.0)
colored2 (3.1.2)
concurrent-ruby (1.1.5)
connection_pool (2.2.2)
cri (2.15.10)
csv (default: 1.0.0)
date (default: 1.0.0)
deep_merge (1.2.1)
did_you_mean (1.2.0)
ed25519 (1.2.4)
erubi (1.9.0)
etc (default: 1.0.0)
facter (4.0.14)
faraday (0.14.0)
faraday_middleware (0.14.0)
fast_gettext (1.1.2)
fcntl (default: 1.0.0)
ffi (1.9.25)
fiddle (default: 1.0.0)
fileutils (default: 1.0.2)
gettext (3.2.9)
gettext-setup (0.31)
gssapi (1.3.0)
gyoku (1.3.1)
hiera (3.6.0)
hiera-eyaml (3.2.0)
highline (1.6.21)
hocon (1.3.0)
httpclient (2.8.3)
io-console (default: 0.4.6)
ipaddr (default: 1.2.0)
jmespath (1.4.0)
json (default: 2.1.0)
little-plugger (1.1.4)
locale (2.1.2)
log4r (1.1.10)
logging (2.2.2)
minitar (0.9)
minitest (5.10.3)
molinillo (0.6.6)
multi_json (1.13.1)
multipart-post (2.1.1)
net-http-persistent (3.1.0)
net-scp (1.2.1)
net-ssh (5.2.0)
net-ssh-krb (0.5.1)
net-telnet (0.1.1)
nori (2.6.0)
openssl (default: 2.1.2)
optimist (3.0.0)
orchestrator_client (0.4.3)
power_assert (1.1.1)
psych (default: 3.0.2)
public_suffix (4.0.3)
puppet (6.14.0)
puppet-resource_api (1.8.12)
puppet-strings (2.4.0)
puppet_forge (2.3.4)
puppetfile-resolver (0.1.0)
r10k (3.4.1)
rake (12.3.0)
rdoc (default: 6.0.1.1)
rgen (0.8.2)
ruby_smb (1.1.0)
rubyntlm (0.6.2)
rubyzip (2.3.0)
scanf (default: 1.0.0)
sdbm (default: 1.0.0)
semantic_puppet (1.0.2)
stringio (default: 0.0.1)
strscan (default: 1.0.0)
terminal-table (1.8.0)
test-unit (3.2.7)
text (1.3.1)
thor (1.0.1)
unicode-display_width (1.7.0)
webrick (default: 1.4.2)
win32-dir (0.4.9)
win32-process (0.7.5)
win32-security (0.2.5)
win32-service (0.8.8)
windows_error (0.1.2)
winrm (2.3.4)
winrm-fs (1.3.4)
xmlrpc (0.3.0)
yard (0.9.24)
zlib (default: 1.0.0)
```

🎉

**Pull request merged**